remix-debug/src/solidity-decoder/types/util.js

```diff
@@ -78,6 +78,8 @@ function toBN (value) {
   } else if (value.indexOf && value.indexOf('0x') === 0) {
     value = ethutil.unpad(value.replace('Ox', ''))
     value = new BN(value === '' ? '0' : value, 16)
+  } else if (value.match(/[0-9a-f]*/)) {
+    value = new BN(value, 16)
```

Comment:
Unsure if I need to `ethutil.unpad` hear too. @yann300, I'm open to suggestions.

Reply:
yes I think you need it here